core/src/test/scala/unit/kafka/server/ApiVersionsRequestTest.scala

```diff
@@ -0,0 +1,41 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package kafka.server
+
+import org.apache.kafka.common.protocol.ApiKeys
+import org.apache.kafka.common.requests.{ApiVersionsRequest, ApiVersionsResponse}
+import org.junit.Assert._
+import org.junit.Test
+
+class ApiVersionsRequestTest extends BaseRequestTest {
+
+  override def numBrokers(): Int = 1
+
+  @Test
+  def testApiVersionsRequest() {
+    val apiVersionsResponse = sendApiVersionsRequest(new ApiVersionsRequest, 0)
+
+    assertEquals("API keys in ApiVersionsResponse must match API keys supported by broker.", ApiKeys.values.length, apiVersionsResponse.apiVersions.size)
+    assertTrue("API versions in ApiVersionsResponse must be supported by broker.", KafkaApis.apiVersionsResponse.apiVersions().containsAll(apiVersionsResponse.apiVersions()))
```

Comment:
I am adding `equals` method to `ApiVersionsResponse.ApiVersion`, not to `ApiVersionsResponse`. Also, I think equals might be useful at other places as well. Having that said if you still strongly feel against having the equals, I can remove it. Let me know.